packages/salesforce-adapter/src/elements_url_retreiver/elements_url_retreiver.ts

```diff
@@ -36,41 +34,11 @@ export const lightiningElementsUrlRetreiver = (baseUrl: URL, elementIDResolver:
   }
   const lightiningUrl = new URL(`${baseUrl.origin.substr(0, suffix.index)}lightning.force.com`)
 
-  const retreiveUrl = async (element: Element): Promise<URL | undefined> =>
-    ((await Promise.all(
-      resolvers.map(resolver => resolver(element, lightiningUrl, elementIDResolver))
-    )).find(values.isDefined))
+  const retreiveUrl = (element: Element): URL | undefined =>
+    (resolvers.map(resolver => resolver(element, lightiningUrl, elementIDResolver))
+      .find(values.isDefined))
```

Comment:
```suggestion
  const retreiveUrl = (element: Element): URL | undefined => (
   wu(resolvers)
     .map(resolver => resolver(element, lightiningUrl, elementIDResolver))
     .find(values.isDefined)
 )
```
adding `wu` here to avoid running all resolvers (it will stop after the first one that returns a value)